public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -251,6 +252,14 @@ export class PanelQueryRunner {
       app,
     } = options;
 
+    if (panelId !== undefined) {
+      const dashboard = getDashboardSrv().getCurrent();
+      const panel = dashboard?.getPanelById(panelId);
+      if (panel) {
+        panel.refreshWhenInView = false;
```

Comment:
Feels like panel state wrapper that sets this can also clear it when it calls this runAllQueries? Or do it in the panel model function runAllQueries?

This lookup logic and state clearing feels a bit messy / out of place

Reply:
I agree it's a bit messy, but if I did it when the panel state wrapper calls `runAllPanelQueries`, or within `runAllPanelQueries` itself, the `refreshWhenInView` value for the panel whose query is being shared would still be true if you refreshed the dashboard when another panel using that query was in view, but the original panel itself is out of view, causing the query to execute when you scroll back up to the original

Reply:
not sure I understand that, this looks to modify the current panel's refreshWhenInView prop not the panel it is referencing in it's shared query. So how come it's different setting 
refreshWhenInView here vs in runAllPanelQueries? is panelId / panel model not the same?

Reply:
Sorry, didn't explain it very well. Say you have the same set up as in the example dashboard json you provided, but we _don't_ set `refreshWhenInView` to false in the panel query runner `run` method as I've done:

1. Dashboard loads, query that is shared and used by other panels is executed. Call this original panel 'A'.
2. Scroll down to the panel that uses the shared query, call this panel 'B'. Panel A is now out of view.
3. Refresh. This sets `refreshWhenInView` to true for panel A.
4. `runSharedRequest` gets executed for panel B, it checks if `!listenToPanel.isInView && listenToPanel.refreshWhenInView` is true (listenToPanel is panel A). This is true, so `listenToRunner.run(modified)` gets executed.
5. Note that panel A's `refreshWhenInView` value is still true.
6. If we scroll back up so that panel A is in view, the query will be needlessly executed again, since `refreshWhenInView` is set.

Reply:
@kaydelaney ok I understand. so you want to set refreshWhenInView to false for the shared panel. 

I moved this to runSharedRequest, it already has the panel there so seems like less complex place to add this to. (also added setting it false to PanelStateWrapper since it's the component that set's it to true

I have one more question, why did you add ` dashboard?.otherPanelInFullscreen(listenToPanel)`  to the if statement in runSharedRequest? 

How can the shared panel be in fullscreen while the query executes for the panel using it?

Reply:
ah, I understand now otherPanelInFullscreen is very confusing :)  i